es/provider.go

```diff
@@ -142,6 +142,7 @@ func Provider() terraform.ResourceProvider {
 			"elasticsearch_opendistro_ism_policy_mapping": resourceElasticsearchOpenDistroISMPolicyMapping(),
 			"elasticsearch_opendistro_monitor":            resourceElasticsearchOpenDistroMonitor(),
 			"elasticsearch_opendistro_roles_mapping":      resourceElasticsearchOpenDistroRolesMapping(),
+			"elasticsearch_snapshot_lifecycle_policy":     resourceElasticsearchSnapshotLifecyclePolicy(),
```

Comment:
This is actually an x-pack feature, can we group it under the xpack resources? We should update the filenames as well.